aspnetcore/mvc/models/model-binding.md

```diff
@@ -1,146 +1,434 @@
+* Converts the string "true" into boolean `true`.
+
+The framework then calls the `GetById` method, passing in 2 for the `id` parameter, and `true` for the `dogsOnly` parameter.
+
+In the preceding example, the model binding targets are method parameters that are simple types. Targets may also be the properties of a complex type. After each property is successfully bound, [model validation](xref:mvc/models/validation) occurs for that property. The record of what data is bound to the model, and any binding or validation errors, is stored in [ControllerBase.ModelState](xref:Microsoft.AspNetCore.Mvc.ControllerBase.ModelState) or [PageModel.ModelState](xref:Microsoft.AspNetCore.Mvc.ControllerBase.ModelState). To find out if this process was successful, the app checks the [ModelState.IsValid](xref:Microsoft.AspNetCore.Mvc.ModelBinding.ModelStateDictionary.IsValid) flag.
+
+## Targets
+
+Model binding tries to find values for the following kinds of targets:
+
+* Parameters of the controller action method that a request is routed to.
+* Parameters of the Razor Pages handler method that a request is routed to. 
+* Public properties of a controller or `PageModel` class, if specified by attributes.
+
+### [BindProperty] attribute
+
+Can be applied to a public property of a controller or `PageModel` class to cause model binding to target that property:
+
+[!code-csharp[](model-binding/samples/2.x/Pages/Instructors/Edit.cshtml.cs?name=snippet_BindProperty&highlight=7-8)]
+
+### [BindProperties] attribute
+
+Available in ASP.NET Core 2.1 and later.  Can be applied to a controller or `PageModel` class to tell model binding to target all public properties of the class:
+
+[!code-csharp[](model-binding/samples/2.x/Pages/Instructors/Create.cshtml.cs?name=snippet_BindProperties&highlight=1-2)]
+
+### Model binding for HTTP GET requests
+
+By default, the `[BindProperty]` and `[BindProperties]` attributes have no effect for HTTP GET requests. Typically, all you need for a GET request is a record ID parameter. The record ID is used to look up the item in the database. Therefore, there is no need to bind a property that holds an instance of the model. In scenarios where you do want properties bound to data from GET requests, set the `SupportsGet` property to `true`:
+
+[!code-csharp[](model-binding/samples/2.x/Pages/Instructors/Index.cshtml.cs?name=snippet_SupportsGet)]
+
+## Sources
+
+By default, model binding gets data in the form of key-value pairs from the following sources in an HTTP request:
+
+1. Form fields 
+1. The request body (For [controllers that have the [ApiController] attribute](xref:web-api/index#binding-source-parameter-inference).)
+1. Route data
+1. Query string parameters
+1. Uploaded files 
+
+For each target parameter or property, the sources are scanned in the order indicated in this list. There are a few exceptions:
+
+* Form fields and request body data are used only for properties of complex types.
```

Comment:
Model binding should be fine reading simple types from forms. If you're seeing otherwise, this might be a bug.